### ISSUE TITLE
test_get_raises_exception_with_full_traceback fails on Python 3.8

### ISSUE THREAD

**Issue by eclipseo** [jodal/pykka#93]
Python 3.8.1 on Fedora Rawhide:
Python 3.8.1, pytest-4.6.9, py-1.8.0, pluggy-0.13.0
```
=================================== FAILURES ===================================
____________ test_get_raises_exception_with_full_traceback[gevent] _____________
runtime = Runtime(name='gevent', actor_class=<class 'pykka.gevent.GeventActor'>, event_class=<class 'gevent._event.Event'>, future_class=<class 'pykka.gevent.GeventFuture'>, sleep_func=<function sleep at 0xf68a6c40>)
    def test_get_raises_exception_with_full_traceback(runtime):
        exc_class_get = None
        exc_class_set = None
        exc_instance_get = None
        exc_instance_set = None
        exc_traceback_get = None
        exc_traceback_set = None
        future = runtime.future_class()
    
        try:
            raise NameError('foo')
        except NameError:
            exc_class_set, exc_instance_set, exc_traceback_set = sys.exc_info()
            future.set_exception()
    
        # We could move to another thread at this point
    
        try:
            future.get()
        except NameError:
            exc_class_get, exc_instance_get, exc_traceback_get = sys.exc_info()
    
        assert exc_class_set == exc_class_get
        assert exc_instance_set == exc_instance_get
    
        exc_traceback_list_set = list(
            reversed(traceback.extract_tb(exc_traceback_set))
        )
        exc_traceback_list_get = list(
            reversed(traceback.extract_tb(exc_traceback_get))
        )
    
        # All frames from the first traceback should be included in the
        # traceback from the future.get() reraise
        assert len(exc_traceback_list_set) < len(exc_traceback_list_get)
        for i, frame in enumerate(exc_traceback_list_set):
>           assert frame == exc_traceback_list_get[i]
E           assert <FrameSummary...ull_traceback> == <FrameSummary ...9 in <module>>
E             -<FrameSummary file /builddir/build/BUILD/pykka-2.0.2/tests/test_future.py, line 89 in test_get_raises_exception_with_full_traceback>
E             +<FrameSummary file /builddir/build/BUILD/pykka-2.0.2/tests/test_future.py, line 89 in <module>>
E             Full diff:
E             - <FrameSummary file /builddir/build/BUILD/pykka-2.0.2/tests/test_future.py, line 89 in test_get_raises_exception_with_full_traceback>
E             ?                                                                                       ^^^^^^^^^^^^^^^^^^^^^^^ ^^^^^^^^  ------ ----
E             + <FrameSummary file /builddir/build/BUILD/pykka-2.0.2/tests/test_future.py, line 89 in <module>>
E             ?                                                                                       ^^ ^    +
tests/test_future.py:115: AssertionError
```

**Comment by jodal** [comment]
Thanks for the bug report!

I cannot reproduce this on Ubuntu 19.10 using Python 3.8.1 built with pyenv. From the paths, it seems this is the output from trying to build a Pykka package from Fedora. Can you reproduce the same test failure if running the test suite on Fedora outside the package build infrastructure?

The steps to do so would be approximately the following:

```
git clone git@github.com:jodal/pykka.git
cd pykka
git checkout v2.0.2
python3.8 -m venv .venv
. .venv/bin/activate
pip install -e ".[dev]"
python -m pytest
```

**Comment by eclipseo** [comment]
Ok I've found the issue, Fedora is shipping a patched gevent 1.4.0 to work with Python 3.8, but it seems the patch is not good enough and it's causing the bug. gevent 1.5a3 solves the issue. I will see if the maintainer is okay with packaging the alpha release.
